### ISSUE TITLE
Scroll through days at plot level

### ISSUE THREAD

**Issue by aeffler** [gwpy/gwsumm#294]
I think this should be easy to implement at the website level. If someone wants to scroll through several days of a certain plot, you have to do it at the page level. I would like to have arrows when I select a plot to look at, such that I can scroll through just that plot as it changes the days.

**Comment by eagoetz** [comment]
I like the idea of this too, but currently have no idea how to implement it, let alone time to dive deep into the guts of how to implement the feature. I'll see what I can do...

**Comment by areeda** [comment]
This kind of display is easy to produce as an animated GIF. It should be possible to add say a week or month look back for specific plots.

However the Summary pages are static html content. It is not possible in the current set up to select an image and specify how many days.

@aeffler please give us an image and time frame you would like to see as a feasibility demo.

**Comment by areeda** [comment]
This is what I do for fun (I know!)

How does [this](https://ldas-jobs.ligo-la.caltech.edu/~joseph.areeda/week.gif) look?
It needs work but should be enough to get the idea.

**Comment by aeffler** [comment]
I think you're doing something way more complicated :) When you click on a
plus currently, there's an arrow on the left and right, and they scroll
through the plots on that particular page. Instead, we want it to stay on
the same plot but go to the next day. To me, it's just a change in the date
of the html address, i do it by hand often in fact. So i fail to understand
why the arrow can't be coded to link to the same url, but with a number
changed in it.

**Comment by aeffler** [comment]
*when you click on a plot

**Comment by eagoetz** [comment]
> So i fail to understand why the arrow can't be coded to link to the same url, but with a number changed in it.

Have you seen the gwsumm codebase? It's quite complicated, but I am trying to understand more so maybe someday you'll get this. Just don't hold your breath.

**Comment by aeffler** [comment]
I was answering JoeA, but like i said "i don't understand" cause i don't.
I'm not holding my breath at all, it's not a priority, just would be a nice
feature.

**Comment by areeda** [comment]
We will discuss this at our next meeting. I see Anamaria's point  but I'm not sure that we do know the name the image are named something like: 
https://ldas-jobs.ligo-la.caltech.edu/~detchar/summary/day/20220722/plots/L1-LOCKED_18A747_TIMESERIES-1342483218-86400.png
I have to check if the image id is fixed and that gps time/duration is always 24hr from UTC 00:00 independent of detector state.

**Comment by eagoetz** [comment]
Yeah, that "18A747" is some SHA256 hash ID. Maybe it is a predictable thing, but I don't know

**Comment by areeda** [comment]
Oh and the animated gif is a fairly simple bash script but let's see if we can do it Anamaria's way. The weekly summary pages generated every day (last 7 days) might dovetail into this.

**Comment by aeffler** [comment]
For me it shows as this:
https://ldas-jobs.ligo-la.caltech.edu/~detchar/summary/day/20220720/#gallery-1
And I just change the date in it...

**Comment by areeda** [comment]
I am more confused than usual. 
How is that different than pressing the '<<' or '>>' buttons next to the date in the tool bar?

**Comment by aeffler** [comment]
When you click on a plot, those buttons go away :)

**Comment by areeda** [comment]
Ah OK. When you click on a plot now the browser is opening "gallery" for me where the left/right arrows move through the images on that page. That seems to be a jQuery object. Does anyone know how to work with those?

**Comment by areeda** [comment]
I think I need to be quiet and let those who know much more about summary page code look into this.

**Comment by eagoetz** [comment]
I can address my own comment on the hash in the filename. Apparently it does not change from day to day, just the absolute path (that includes the date string) and the GPS time of the plot. The tricky thing may be when there are leap seconds so that GPS time and UTC time may get offset by one second relative to each other. We'll see how this could be done, though it will take some dev work in gwsumm/gwdetchar packages.

**Comment by duncanmmacleod** [comment]
> I can address my own comment on the hash in the filename. Apparently it does not change from day to day, just the absolute path (that includes the date string) and the GPS time of the plot. The tricky thing may be when there are leap seconds so that GPS time and UTC time may get offset by one second relative to each other. We'll see how this could be done, though it will take some dev work in gwsumm/gwdetchar packages.

The hash is taken from the type of plot and the list of channels included, see, for example:

https://github.com/gwpy/gwsumm/blob/2050fcabbb0d8ab09534fb844d8af0d4e289e648/gwsumm/plot/core.py#L331-L341
